evm/src/executor/inspector/cheatcodes/ext.rs

```diff
@@ -464,18 +465,17 @@ fn write_json(
     path: impl AsRef<Path>,
     json_path_or_none: Option<&str>,
 ) -> Result<Bytes, Bytes> {
-    let json: Value = serde_json::from_str(object).unwrap_or(Value::String(object.to_owned()));
+    let json: Value =
+        serde_json::from_str(object).unwrap_or_else(|_| Value::String(object.to_owned()));
```

Comment:
Why have a map here?

Reply:
This change is a byproduct of cleaning up some clippy warnings:
![Screenshot from 2022-11-26 13-45-55](https://user-images.githubusercontent.com/8406232/204104392-82437c9a-4bfb-4d5b-98a1-a82544df2826.png)

Reply:
hm, curious about why clippy would prefer a map. @mattsse, any idea?

Reply:
According to the [big book o' clippy lints](https://rust-lang.github.io/rust-clippy/master/index.html#or_fun_call), it probably has something to do with allocating heap memory for the `object` str slice via `to_owned`, but not 100% sure. It likely is an unnecessary change given that it's not very expensive.